src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -118,19 +117,18 @@ private static void CallPropertyAddRange<TElement>(object target, object source)
 
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
-            var propertyHelpers = PropertyHelper.GetProperties(controller);
+            var controllerMetadata = _modelMetadataProvider.GetMetadataForType(containerType);
```

Comment:
> Are you suggesting this code should assume the top-level property does not also initialize the hidden one? Sounds like a lot of complication, especially given issues when base.Property would be property initialized as-is.

Yes. It's unlikely anyone would hit this bug because activation only really applies to `ActionContext` and `ActionBindingContext`.